e2e-test-server/src/main/java/com/google/cloud/opentelemetry/endtoend/Server.java

```diff
@@ -0,0 +1,139 @@
+ *   <li>Setting up a publisher queue for outbound "RPC Response" messages
+ *   <li>Converting from outbound {@link Response} to pubsub messages.
+ *   <li>Handling any/all failures escaping the test scenario.
+ * </ul>
+ *
+ * <p>This class includes a main method which runs the integration test driver using locally
+ * available credentials to acccess pubsub channels.
+ */
+public class Server implements AutoCloseable {
+  private final ScenarioHandlerManager scenarioHandlers = new ScenarioHandlerManager();
+  private final Publisher publisher;
+  private final Subscriber subscriber;
+
+  public Server() throws Exception {
+    this.publisher = Publisher.newBuilder(Constants.getResponseChannel()).build();
+    this.subscriber =
+        Subscriber.newBuilder(Constants.getRequestSubscription(), this::handleMessage).build();
+    subscriber.addListener(
+        new Subscriber.Listener() {
+          @Override
+          public void failed(Subscriber.State from, Throwable failure) {
+            // Handle failure. This is called when the Subscriber encountered a fatal error and is
+            // shutting down.
+            System.err.println(failure);
+          }
+        },
+        MoreExecutors.directExecutor());
+  }
+
+  /** Starts the subcriber pulling requests. */
+  public void start() {
+    subscriber.startAsync().awaitRunning();
+  }
+
+  /** Closes our subscriptions. */
+  public void close() {
+    if (subscriber != null) {
+      subscriber.stopAsync();
+    }
+    if (publisher != null) {
+      publisher.shutdown();
+    }
+  }
+
+  /** This method converts from {@link Response} to pubsub and sends out the publisher channel. */
+  public void respond(final String testId, final Response response) {
+    final PubsubMessage message =
+        PubsubMessage.newBuilder()
+            .putAttributes(Constants.TEST_ID, testId)
+            .putAttributes(Constants.STATUS_CODE, Integer.toString(response.statusCode().ordinal()))
+            .setData(response.data())
+            .build();
+    ApiFuture<String> messageIdFuture = publisher.publish(message);
+    ApiFutures.addCallback(
+        messageIdFuture,
+        new ApiFutureCallback<String>() {
+          public void onSuccess(String messageId) {}
+
+          public void onFailure(Throwable t) {
+            System.out.println("failed to publish response to test: " + testId);
+            t.printStackTrace();
+          }
+        },
+        MoreExecutors.directExecutor());
+  }
+
+  /** Execute a scenario based on the incoming message from the test runner. */
+  public void handleMessage(PubsubMessage message, AckReplyConsumer consumer) {
+    if (!message.containsAttributes(Constants.TEST_ID)) {
+      consumer.nack();
+      return;
+    }
+    String testId = message.getAttributesOrDefault(Constants.TEST_ID, "");
+    if (!message.containsAttributes(Constants.SCENARIO)) {
+      respond(testId, Response.invalidArugment("Exepcted attribute \"{SCENARIO}\" is missing"));
```

Comment:
copypasta artifact from python

```suggestion
      respond(testId, Response.invalidArugment(String.format("Expected attribute \"%s\" is missing", Constants.SCENARIO)));
```